src/System.Net.Sockets/tests/FunctionalTests/KeepAliveTest.netcoreapp.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class KeepAliveTest
+    {
+        private bool RunningOnWindowsBelow10v1703
+        {
+            get
+            {
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    return false;
+                Version version = Environment.OSVersion.Version;
+                return version.Major < 10 || version.Major == 10 && version.Build < 15063;
+            }
+        }
```

Comment:
There are already helpers in the test PlatformDetection class for this, e.g. `!PlatformDetection.IsWindows10Version1703OrGreater`.

Reply:
(it's not exactly the same as yours with regards to platform, so your call sites would need to only use this if on Windows... on Unix, e.g., IsWindows10Version1703OrGreater will return false.